File: src/main/java/yuuki1293/panmodern/datagen/DataGenerators.java
package yuuki1293.panmodern.datagen;

import net.minecraft.data.DataGenerator;
import net.minecraft.data.PackOutput;
import net.neoforged.neoforge.data.event.GatherDataEvent;

public class DataGenerators {

    public static void gatherData(GatherDataEvent event) {
        DataGenerator generator = event.getGenerator();
        PackOutput packOutput = generator.getPackOutput();

        generator.addProvider(event.includeClient(), new PANModernLanguageProvider(packOutput));
        generator.addProvider(
            event.includeClient(),
            new PANModernBlockStateProvider(packOutput, event.getExistingFileHelper()));
    }
}

File: src/main/kotlin/yuuki1293/panmodern/datagen/DataGenerators.kt
package yuuki1293.panmodern.datagen

import net.minecraft.data.DataGenerator
import net.minecraft.data.PackOutput
import net.neoforged.neoforge.data.event.GatherDataEvent

object DataGenerators {
    fun gatherData(event: GatherDataEvent) {
        val generator: DataGenerator = event.generator
        val packOutput: PackOutput = generator.packOutput

        generator.addProvider(event.includeClient(), PANModernLanguageProvider(packOutput))
        generator.addProvider(
            event.includeClient(),
            PANModernBlockStateProvider(packOutput, event.existingFileHelper),
        )
    }
}
